### DISCUSSION TITLE
How do you add a new language syntax highlighter to codehike?

### DISCUSSION THREAD

**Discussion by lancejpollard** [code-hike/codehike#324] [Q&A]
I am using codehike with mdx and Next.js because of [this](https://blog.anishde.dev/powerful-code-blocks-with-code-hike-and-mdx) wonderful blog post. But I would like to syntax highlight the Coq language, which [doesn't appear to be supported](https://codehike.org/docs/codeblocks). [Here](https://github.com/github/linguist/tree/master/vendor/grammars) are a bunch of grammars, like [this Coq one](https://github.com/mkolosick/Sublime-Coq/blob/0d1f70727b19007e55780bcd7b74f70c399eba3b/Coq.tmLanguage) (it is very short, nothing fancy yet).

How can I add support for this language to my Next.js project? That is, how can I just hack it in without making a PR to `codehike` to add formal support? Or how can I also create the PR to add formal support to codehike?

Thank you!

**Comment by pomber** [comment]
Hi, Code Hike now uses [lighter](https://github.com/code-hike/lighter) for highlighting. 

If you just want to try your grammar with codehike, the easiest way would be to replace one of the existing grammars in `node_modules/@code-hike/lighter/grammars` with your own. For example, replace the content of abap.tmLanguage.json with the content of coq.tmLanguage.json (you need a json version of the grammar), and then use `abap` as the language.

That said, we plan to support custom grammars in the future: https://github.com/code-hike/codehike/issues/297
